test: Add test case for reference regional target tcp proxy in CFR

gemmahou opened the pull request:
### Change description

<!--
Describe what this pull request does.

* If your pull request is to address an open issue, indicate it by specifying the
issue number: 

For example: "Fixes #858"
-->
based on #3047, #3112

Add additional test case for compute forwarding rule, verify that it can now reference to a regional tcp proxy

### Tests you have done

<!--

Make sure you have run "make ready-pr" to run required tests and ensure this PR is ready to review. 

Also if possible, share a bit more on the tests you have done. 

For example if you have updated the pubsubtopic sample, you can share the test logs from running the test case locally.

go test -v -tags=integration ./config/tests/samples/create -test.run TestAll -run-tests pubsubtopic

-->

- [N/A] Run `make ready-pr` to ensure this PR is ready for review.
- [x] Perform necessary E2E testing for changed resources.

yuwenma commented on apis/compute/v1beta1/targettcpproxy_types.go:
```diff
@@ -87,7 +87,7 @@ type ComputeTargetTCPProxyObservedState struct {
 // +kubebuilder:resource:categories=gcp,shortName=gcpcomputetargettcpproxy;gcpcomputetargettcpproxies
 // +kubebuilder:subresource:status
-// +kubebuilder:metadata:labels="cnrm.cloud.google.com/tf2crd=true";"cnrm.cloud.google.com/stability-level=stable";"cnrm.cloud.google.com/managed-by-kcc=true";"cnrm.cloud.google.com/system=true"
+// +kubebuilder:metadata:labels="cnrm.cloud.google.com/stability-level=stable";"cnrm.cloud.google.com/managed-by-kcc=true";"cnrm.cloud.google.com/system=true"
```
Let's not turn off the TF-based controller in this PR

gemmahou replied:
Done. Let's enable regional TCP Proxy first #3323 before merging this PR.

yuwenma reviewed (commented):
/lgtm
/approve
/hold

one suggestions

gemmahou commented on pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/regionalcomputeforwardingruletcp/create.yaml:
```diff
@@ -0,0 +1,35 @@
+# limitations under the License.
+
+apiVersion: compute.cnrm.cloud.google.com/v1beta1
+kind: ComputeForwardingRule
```
We have it turned on in test: https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/811204fcd3c44b3d7a224640cb25700c0724e34e/dev/tasks/run-e2e#L29

Let me remove this and add direct annotation in test, to avoid confusions.

gemmahou replied:
Well, I feel it's better to keep it in run-e2e until we enable direct controller for CFR. Since we are going to enable it soon, we do not need to duplicate those other verified CFR tests. Back to your original question, this specific test(CFR with regional target TCP Proxy) in this PR is using direct controller. WDYT?

yuwenma replied:
ah. I see.

gemmahou commented:
I'll remove the hold and get this merged, since we have test coverage for direct CFR with regional target TCP proxy. If we want to remove the KCC_USE_DIRECT_RECONCILERS flag and duplicate all CFR tests(add original test and -direct test), I feel it's might be clearer to add them in a separate PR.

gemmahou commented:
I removed the KCC_USE_DIRECT_RECONCILERS flag for CFR and use annotation in test yaml. Also added both TF and direct test case for CFR with regional target tcp proxy. For other existing direct tests, we can duplicate them without direct annotation in another PR, if needed.

yuwenma commented on pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/globalcomputeforwardingruletcp-direct/_http.log:
```diff
@@ -424,10 +424,9 @@ X-Xss-Protection: 0
 Content-Type: application/json
-User-Agent: kcc/controller-manager
-x-goog-request-params: project=${projectId}&target_tcp_proxy=${targetTcpProxyID}
+User-Agent: Terraform/ (+https://www.terraform.io) Terraform-Plugin-SDK/2.10.1 terraform-provider-google-beta/kcc/controller-manager
```
This means the TCPProxy resource is changed from direct to TF-based. Could you investigate what is missing?

yuwenma replied:
ok, I think I'm confused because this test name is "-direct".

yuwenma commented on pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/regionalcomputeforwardingruletcp-direct/_http.log:
Could you upload the real GCP log for this test suite?

gemmahou replied:
https://paste.googleplex.com/6307517609803776

No changes to golden object log, some changes to http log but nothing related to CFR

yuwenma commented on pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computeforwardingrule/regionalcomputeforwardingruletcp/_generated_object_regionalcomputeforwardingruletcp.golden.yaml:
ditto: Could you upload the real GCP Log for this test suite?

gemmahou replied:
This test is for TF-based CFR with direct regional TCP proxy, here's the diff: https://paste.googleplex.com/5134954728783872

direct regional TCP proxy also covered in its own tests.

yuwenma replied:
I see. the past seems to only contain partial differences. Do you know which resource has those diff?

yuwenma reviewed (commented):
/lgtm
/approve

yuwenma commented:
/lgtm
